### ISSUE TITLE
Is there interest for X stuff in the basement?

### ISSUE THREAD

**Issue by Ferruck** [BobBuildTool/basement#107]
While compiling GTK I had to compile a lot of X libraries as dependencies. Is there general interest in having them in the basement? Specifically I have recipes for:

* libx11
* libxau
* libxcb
* libxcursor
* libxdamage
* libxext
* libxfixes
* libxinerama
* libxi
* libxrandr
* libxrender
* libxshmfence
* libxtrans
* libxxf86vm
* libpciaccess (sic!)
* xcb-proto
* xorgproto
* xorg-macros

Related, but not exactly X stuff:
* xkbcommon

I have also adapted the `libs::cairo` recipe to include Xlib support, which could, in turn, also be provided to the basement.

**Comment by daxcore** [comment]
IMO it would be a good solution to create another layer repository for X!
I guess there are people how would use X, if it is available.
It is a hugh benefit of/for bob, if we share our recipes AFAP.

**Comment by jkloetzke** [comment]
I think @daxcore has a point. I wouldn't create a dedicated X-Layer but we should probably create a "Linux" layer and restrict the basement layer to toolchains, common build tools like autotools/cmake/meson... and the sandbox. Despite putting it in a different layer, I'm very much interested in having this stuff upstream. :smiley:

**Comment by Ferruck** [comment]
I'm fine with that. How to proceed from here, @jkloetzke? Will you create a new repository where I open a MR for the changes?

**Comment by jkloetzke** [comment]
Yes. Probably there are already a couple of packages in basement that shall be moved immediately there...

**Comment by Ferruck** [comment]
Any plans when to tackle this?

**Comment by jkloetzke** [comment]
Yes, but it will take some time. I'm currently occupied with other stuff.

**Comment by mahaase** [comment]
now i'm interested into this too :) 
i want to try to integrate qt6 with X stuff.

**Comment by jkloetzke** [comment]
I've created a proposal in #115. It's completely untested (I'm especially unsure about `devel::makedepend` removal from the sandbox). I would create two new layers: the obvious `gnu-linux` layer but also a `win32-bin` layer for the binary packages of win32 tools. Of course these would be moved to dedicated repos once it's clear that the structure is ok.

**Comment by Ferruck** [comment]
I've just come back to work so I'll have a look at this as soon as I find some time.

**Comment by mahaase** [comment]
@jkloetzke this one is not fully done by the commit that closes this issue.
the x libraries are still not available, but the prerequisites are just done.

**Comment by jkloetzke** [comment]
Right. And I even did not yet create the other repositories... :see_no_evil:

**Comment by mahaase** [comment]
push. still interessted into the x11 stuff 😅
